### PR TITLE
feat(dotfiles): capture changed copy targets

### PR DESCRIPTION
**Pull request by jdx**
## Summary
- add `mise bootstrap dotfiles add --changed` to capture all drifted copy-mode targets
- preserve dry-run and transactional multi-target capture behavior
- exclude symlinks, templates, and inline content that do not need or cannot support reverse capture
- document and cover bulk capture and argument validation

## Testing
- `mise run lint-fix`
- `mise run test:e2e e2e/cli/test_dotfiles_files`

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

---

> [!NOTE]
> **Medium Risk**
> Bulk reverse capture writes many source files from live targets and enforces trust checks; mistakes or untrusted config could copy wrong content, though directory/template/symlink exclusions limit blast radius.
> 
> **Overview**
> Adds **`mise bootstrap dotfiles add --changed`** to push live edits back into dotfile sources for every **copy-mode regular file** that currently **differs** from its source—without naming each path.
> 
> **CLI behavior:** `TARGET` arguments are optional when `--changed` is used; running `add` with no targets and no `--changed` errors clearly, and `--changed` cannot be combined with explicit targets. Bulk selection skips directory copies (additive/exclude semantics), symlink targets, templates, and inline content. Non-global entries require **trusted** configuration before bulk capture; safe mode blocks untrusted bulk writes.
> 
> Existing per-target capture, dry-run, and transactional multi-target `add` behavior stay the same; `edit` passes `changed: false` into `DotfilesAdd`. Docs, usage spec, man page, and e2e tests cover bulk capture, edge cases, and validation.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 342e4cbee1d86981e6371c380565c2283c22f406. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

- **New Features**
  - Added a `--changed` option to capture all modified copy-mode dotfiles at once.
  - Supports dry runs and confirmation-based updates.
  - Targets are now optional when using the add command.
- **Bug Fixes**
  - Running the add command without a target or `--changed` reports a clear error.
  - Prevented combining `--changed` with explicit targets.
  - Re-running after capturing changes correctly reports no remaining changes.
  - Safely excludes unsupported dotfile types and untrusted configurations.
- **Documentation**
  - Added command examples and guidance for capturing changed dotfiles.

### CODE REVIEW

File: src/cli/dotfiles/add.rs [hunk 1]
```diff
@@ -79,6 +89,20 @@ impl DotfilesAdd {
...
+                    && system::files::check(&config, req)? != system::files::FileState::Applied
+                {
+                    self.targets.push(req.target_raw.clone());
+                }
+            }
+            if self.targets.is_empty() {
+                info!("dotfiles: no changed copy-mode targets");
+                return Ok(());
+            }
+        }
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/dotfiles/add.rs:119]: Addressed in 15cc8b7ee: bulk selection now accepts only global/system or independently trusted declaring configs; safe mode with an untrusted project config fails before capture. Added an e2e regression test.\n\n*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: src/cli/dotfiles/add.rs [hunk 2]
```diff
@@ -79,6 +89,20 @@ impl DotfilesAdd {
...
         let config = Config::get().await?;
         let managed = system::files::files_from_config(&config)?;
+        if self.changed {
+            for req in &managed {
+                if req.mode == FileMode::Copy
+                    && req.target.exists()
+                    && system::files::check(&config, req)? != system::files::FileState::Applied
+                {
+                    self.targets.push(req.target_raw.clone());
+                }
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/dotfiles/add.rs:112]: Addressed in 15cc8b7ee: --changed now selects regular copy-mode files only. Additive directory copies are excluded, and the e2e test verifies an excluded source-only file survives a bulk-capture attempt.\n\n*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: src/cli/dotfiles/add.rs [hunk 3]
```diff
@@ -79,6 +90,29 @@ impl DotfilesAdd {
...
         let config = Config::get().await?;
         let managed = system::files::files_from_config(&config)?;
+        if self.changed {
+            for req in &managed {
+                if req.mode == FileMode::Copy
+                    && req.target.is_file()
+                    && !req.target.is_symlink()
+                    && !req.source.is_dir()
+                    && system::files::check(&config, req)? != system::files::FileState::Applied
+                {
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/dotfiles/add.rs:103]: Addressed in 342e4cbee: bulk capture now accepts only `FileState::Differs(_)`, so a missing source whose former directory shape cannot be proven is left for explicit `add <target>`. The e2e test covers a deleted directory source replaced by a regular target file.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: src/cli/dotfiles/add.rs [hunk 4]
```diff
@@ -79,6 +90,29 @@ impl DotfilesAdd {
...
+                            req.origin.config.display_user()
+                        );
+                    }
+                    self.targets.push(req.target_raw.clone());
+                }
+            }
+            if self.targets.is_empty() {
+                info!("dotfiles: no changed copy-mode files");
+                return Ok(());
+            }
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/dotfiles/add.rs:118]: Addressed in 342e4cbee: the empty-selection path now calls `warn_if_dotfiles_ignored()` before reporting no changed files. Added an ignored-config regression assertion.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
